core-bundle/src/EventListener/DataContainer/TemplateOptionsListener.php

```diff
@@ -46,7 +46,10 @@ public function __invoke(DataContainer $dc): array
         }
 
         $type = $dc->getCurrentRecord()['type'] ?? null;
-        $defaultTemplate = $this->customTemplates[$type] ?? $this->getLegacyDefaultTemplate($dc);
+
+        if (null !== $type) {
+            $defaultTemplate = $this->getLegacyDefaultTemplate($type) ?? $this->customTemplates[$type] ?? null;
```

Comment:
How can this be correct? Now the legacy template takes precedence over the custom templates.